crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -167,15 +179,22 @@ where
     }
 }
 
+pub enum ParamConflictType {
+    Query,
+    Resource,
+    Default,
```

Comment:
I'm not immediately clear on what this enum variant would be used for.

Reply:
Rather, I can see where it's used, but I think it should have a clearer name (or documentation explaining that it's for "other" conflicts.

Reply:
yea, im not too sure about this part. I might consider just having a bool for whether it should display a query specific message or not. The Resource one is not really very different to the Default. Another possibility might be having SystemParamState::init return a result instead, which could be made more descriptive (but that would be in another pr).

Reply:
I decided to remove it in favour of using a bool instead